qadence/constructors/feature_maps.py

```diff
@@ -1,106 +1,224 @@
 from __future__ import annotations
 
-from typing import Type, Union
+import inspect
+import math
+from typing import Callable, Type, Union
 
-import numpy as np
 import sympy
 
 from qadence.blocks import AbstractBlock, KronBlock, chain, kron, tag
-from qadence.operations import RX, RY, RZ, H
+from qadence.logger import get_logger
+from qadence.operations import PHASE, RX, RY, RZ, H
 from qadence.parameters import FeatureParameter, Parameter
+from qadence.types import BasisSet, ReuploadScaling, TParameter
 
-Rotation = Union[RX, RY, RZ]
+logger = get_logger(__name__)
+
+TRotation = Type[Union[RX, RY, RZ, PHASE]]
```

Comment:
Having a list of the accepted objects and then doing `Type[list]` or `Type[Union[list]]` doesn't work unfortunately. I have tried a few variations in other modules but haven't quite figured out what the correct syntax is.

Reply:
hm, OK. Another silly question, wouldn't Python `type` work instead of `typing.Type` ?

Reply:
Yes that does work

Reply:
OK I would go for that since we tend to avoid using `typing` as much as possible. There are few other places where you still used `typing` that I think worth revisiting iirc.

Reply:
Yes in fact I was now googling the differences between `Type` and `type` and apparently `Type` is deprecated since python 3.9 and just returns `type`: https://docs.python.org/3/library/typing.html#typing.Type

Reply:
And apparently type hinting with `Callable` should be done from `collections.abc` 🤔 https://docs.python.org/3/library/typing.html#typing.Callable

Reply:
Had to revert to `Type` due to a bug that causes the pipeline in `3.9` to fail (that's fixed in `3.10`).

Reply:
😞 OK makes sense

Reply:
Sorry my bad the bug is with `Union` vs `|` and not `type`